workspaces/orchestrator/plugins/orchestrator-form-react/src/utils/getSortedStepEntries.ts

```diff
@@ -47,6 +49,40 @@ export const getSortedStepEntries = (
     });
   }
 
+  // Filter out hidden steps (fields with ui:hidden: true)
+  // Also filter out steps where ALL inputs are hidden
+  sortedStepEntries = sortedStepEntries.filter(([_, subSchema]) => {
```

Comment:
IIUC this feature does not make sense for an entire step
Notice that if you hide the entire step it won't be rendered at all, and this is not the purpose of this feature

Reply:
Do the widgets in a hidden step still need to be functional? If so, I'm not sure this requirement is met in this implementation.